src/server.c

```diff
@@ -4124,6 +4125,14 @@ int processCommand(client *c) {
         }
     }
 
+    /* Check if the command has cross slot keys, incompatible with cluster mode. */
+    if (server.cluster_compatibility_sample_ratio && !server.cluster_enabled &&
+        !(!(c->cmd->flags&CMD_MOVABLE_KEYS) && c->cmd->key_specs_num == 0) &&
+        SHOULD_CLUSTER_COMPATIBILITY_SAMPLE())
+    {
+        checkCommandKeysCrossSlot(c);
```

Comment:
i'm not sure i understand how is multi-exec handled?

Reply:
oh, i missed, i didn't check all command in  multi-exec, just only one by one

Reply:
done